qubesbuilder/executors/qubes.py

```diff
@@ -167,28 +176,41 @@ def run(  # type: ignore
                     self.replace_placeholders(str(f))
                     for f in files_inside_executor_with_placeholders
                 ]
+                sed_rhs = (
+                    str(self.get_builder_dir())
+                    .replace("@", "x@BUILDER_DIR@x")
+                    .replace("\\", "\\\\")
+                    .replace("&", "\\&")
+                    .replace("#", "\\#")
+                    .replace("\n", "\\\n")
+                )
                 sed_cmd = [
-                    f"sed -i 's#@BUILDER_DIR@#{self.get_builder_dir()}#g' {' '.join(files)}"
+                    f"sed -i -- 's#@BUILDER_DIR@#'{quote(sed_rhs)}'#g; s#x@BUILDER_DIR@x#@#g' {' '.join(map(quote, files))}"
                 ]
                 if sed_cmd:
-                    sed_cmd = [c.replace("'", "'\\''") for c in sed_cmd]
                     sed_cmd = [
                         "/usr/bin/qvm-run-vm",
+                        "--",
                         dispvm,
-                        f'bash -c \'{" && ".join(sed_cmd)}\'',
+                        sed_cmd[0],
```

Comment:
maybe make `sed_cmd` a string variable instead of array that has further elements ignored?